### PR TITLE
Cobra based autocompletion

### PR DESCRIPTION
**Pull request by sdowell**
Prior to this change, the kpt CLI exposed two methods for enabling shell
autocompletion:
- kpt --install-completion
- kpt completion [bash|fish|zsh|powershell]

The former was implemented using complete, whereas the latter is
provided by Cobra as a default. There is not a clear motivation to
offer two methods to provide autocompletion functionality, so this
change aims to consolidate these into a single method.

This change removes the --install-completion option and applies some
fixes to enable proper operation of Cobras autocompletion. The decision
was made to favor the Cobra implementation over the custom one in order
to reduce maintenance cost and offer similar behavior to other popular
CLIs which leverage Cobra (e.g. kubectl)

### CODE REVIEW

File: site/installation/README.md [hunk 1]
```diff
@@ -27,6 +27,47 @@ Verify the version:
...
+#### Enable kpt completion for all your shell sessions
+bash:
+```shell
+echo 'source <(kpt completion bash)' >> ~/.bashrc
+```
+zsh:
+```shell
+echo 'source <(kpt completion zsh)' >> ~/.zshrc
+```
+etc.
```
Comment (droot) [inline site/installation/README.md:69]: This is great.

I started wondering if we can help users discover the shell automatically. This posts mentions a few ways to get the current shell https://askubuntu.com/questions/590899/how-do-i-check-which-shell-i-am-using

`$0` seems to be contain the shell. See if we can use this ?

**Review by droot — APPROVED** [review]
This is excellent work @sdowell . I am happy to get this merged this as-is. I have another suggestion to explore but can be done as a follow up if that further helps the users.

File: site/installation/README.md (hunk 1 above)
Comment (sdowell) [reply site/installation/README.md:69]: Thank you for the feedback 😁 

Just to clarify:
Are you suggesting that we include an example in the docs on how to determine the current shell, or that we change the kpt CLI behavior to try to dynamically discover/infer which shell the user wants updated?

Comment (droot) [reply site/installation/README.md:69]: I was leaning towards `we change the kpt CLI behavior to try to dynamically discover/infer which shell the user wants updated` as the default behavior. So for ex.

```sh
# running the following command configures the autocompletion currently active shell 
source <(kpt completion)

# if one wants to configure auto-completion for shell other than the currently active one
kpt completion <your-fav-shell>
```

Comment (sdowell) [reply site/installation/README.md:69]: To summarize our offline discussion:

As the current behavior is the built-in/default behavior of Cobra, and is also the behavior of other cobra-based command line tools (such as [kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl-linux/#enable-shell-autocompletion)) - we will leave this as-is for now

File: site/installation/README.md [hunk 2]
```diff
@@ -27,6 +27,47 @@ Verify the version:
 $ kpt version
 ```
 
+## (Optional) enable shell auto-completion
```
Comment (phanimarupaka) [inline site/installation/README.md:30]: Can we move this to a separate PR and merge it after this change is released. Else the docs will be ahead of release as they are released with PR merge.

**Review by phanimarupaka — APPROVED** [review]

File: site/installation/README.md (hunk 2 above)
Comment (sdowell) [reply site/installation/README.md]: Done. The docs change has been removed, and I will shortly create a new PR with just the docs change
